modules/openapi-generator/src/main/resources/python/configuration.mustache

```diff
@@ -256,51 +261,51 @@ class Configuration(object):
 
         :return: The Auth Settings information dict.
         """
-        return {
+        auth = {}
 {{#authMethods}}
 {{#isApiKey}}
-            '{{name}}':
-                {
-                    'type': 'api_key',
-                    'in': {{#isKeyInCookie}}'cookie'{{/isKeyInCookie}}{{#isKeyInHeader}}'header'{{/isKeyInHeader}}{{#isKeyInQuery}}'query'{{/isKeyInQuery}},
-                    'key': '{{keyParamName}}',
-                    'value': self.get_api_key_with_prefix('{{keyParamName}}')
-                },
+        if self.api_key:
```

Comment:
Why not add a condition to only set this if the key is in api_key?
Something like if `self.api_key and '{{keyParamName}}' in self.api_key`?

Reply:
Sure. BTW, is there a good explanation of the api key and api key prefix? It's a bit confusing and I can add documentation. For example, somebody who reads the OAS cookie example at https://swagger.io/docs/specification/authentication/cookie-authentication/ may not be clear how api_key and api_key_prefix are supposed to be set. Initially I thought it had to be
self.api_key["JSESSIONID"]="abcde12345", but that's not how it works.

Reply:
I don't know exactly how the api_key vs api_key_prefix work.
The best way to learn about it would be to read the current code and the PRs the created/edited those features.
Feel free to add comments describing those variables in the code

Reply:
I found out how api keys work. For example, given the following security scheme named 'cookieAuth' in the OpenAPI specification:

```
      components:
        securitySchemes:
          cookieAuth:         # name for the security scheme
            type: apiKey
            in: cookie
            name: JSESSIONID  # cookie name
```

The client should configure the cookie like this:
```
       configuration = openapi.Configuration(
          api_key={'JSESSIONID': the_cookie_value},
          api_key_prefix={'JSESSIONID': 'JSESSIONID='})
```

But I think there is a bug at line https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/python/configuration.mustache#L241

There should be no space: 
```
return "%s %s" % (prefix, key)
```

Replace with:
```
return "%s%s" % (prefix, key)
```

Otherwise there will be an extra space in the cookie in the HTTP request.
Cookie: JSESSIONID=abcde\r\n

Reply:
I have added a code sample.